Add Twilio SMS test script and credential setup

bootupAbdullah opened the pull request:
- Created a new Twilio account and generated test credentials (Account SID, Auth Token, phone number)

- Stored credentials locally using .env (test environment only)

- Built a standalone script (send-test-sms.js) to confirm message delivery

- Successfully sent an SMS and confirmed delivery

bootupAbdullah committed:
add dockerfile fixes and initial Twilio SMS test script

bootupAbdullah committed:
Add Twilio SMS test script and Dockerfile setup fixes

SAUMILDHANKAR commented:
- Closes #8 
- Why
- How
- Before After screenshot

peanutenthusiast commented on send-test-sms.js:
```diff
@@ -0,0 +1,20 @@
+require('dotenv').config();
+const twilio = require('twilio');
+
+const client = twilio(
+  process.env.TWILIO_ACCOUNT_SID,
+  process.env.TWILIO_AUTH_TOKEN
+);
```
you could reference these env values from here: 
https://github.com/OpenSourceFellows/chatbot/blob/main/src/config/config.js

antaerus replied:
Since the above linked config file already validates and loads these values, reference the above file's properties at the top of the file instead of accessing `process.env` directly. This gives you validated, typed values rather than raw environment strings.

like so --
```
const { twilioAccountSid, twilioAuthToken} = require('path/to/config')
```

SAUMILDHANKAR reviewed (commented):
Great job @bootupAbdullah on creating your first PR for this project. I was reading up the issue and I understand we need to make sure twilio credentials created in this PR can be used by others in this project going forward. So, maybe we need to store them securely somewhere. Would be great if we can document that information with this PR. Thank you.

bootupAbdullah committed:
Resolve merge conflict in pnpm-lock.yaml

bootupAbdullah commented:
✅ Resolved all merge conflicts — aligned with latest ESLint 9.x ecosystem. Ready for review and merge.

antaerus commented on send-test-sms.js:
```diff
@@ -0,0 +1,20 @@
+client.messages
+  .create({
+    body: 'Test Message: Abdullah, Project Earth, today is June 9th, time is 5:22 PM EST',
+    from: process.env.TWILIO_PHONE_NUMBER,
```
see above for suggestion of importing relevant property from config file

bootupAbdullah replied:
This is great, thank you.

bootupAbdullah replied:
✅ Updated send-test-sms.js to use the shared config module for Twilio credentials as requested
✅ Successfully tested message delivery locally
✅ Merge conflicts resolved (mostly ESLint version upgrades)
⚠️ Non-blocking .env validation warning (POSTGRES_HOST) still logs, but does not affect SMS
Ready for final review and merge — let me know if anything else is needed!

antaerus replied:
@bootupAbdullah - thanks for your hard work. Would you mind checking the merge conflicts?

antaerus reviewed (changes requested):
Looks good. Consider the suggestions re: reference of environment variables.

antaerus commented:
@bootupAbdullah FYI on docker COPY vs ADD
https://stackoverflow.com/questions/24958140/what-is-the-difference-between-the-copy-and-add-commands-in-a-dockerfile

SAUMILDHANKAR commented:
> Thanks again for the review!
> 
> Just to clarify: The Twilio credentials used in this PR are test credentials generated through my own Twilio developer account, as per guidance shared in Slack. These are for local development only and not meant to be reused or committed.
> 
> ## 📄 Setup for future contributors
> The `.env.example` file includes placeholders:
> 
> ```dotenv
> TWILIO_ACCOUNT_SID=your-account-sid
> TWILIO_AUTH_TOKEN=your-auth-token  
> TWILIO_PHONE_NUMBER=your-twilio-number
> ```
> 
> ## 📌 Instructions included in PR description
> * Create Twilio account
> * Obtain credentials
> * Use `send-test-sms.js` script for verification
> 
> ## Per Slack guidance from Ngole:
> > "Each developer is expected to create their own test credentials. These are used to verify that your feature works as expected in local development. When the app is ready for production, maintainers will generate the appropriate secure credentials."
> 
> Please let me know if you'd like me to add this information to the project documentation as well!

Thanks @bootupAbdullah yes it will be great to add this info to documentation. Please let us know the information required to create credentials for the organization and maybe we can save them in a secure place. Thank you.

bootupAbdullah committed:
Refactor: use config module for Twilio credentials in send-test-sms.js

antaerus commented on send-test-sms.js:
```diff
@@ -0,0 +1,17 @@
+const twilio = require('twilio');
+const { twilioAccountSID, twilioAuthToken, twilioPhoneNumber } = require('./src/config/config');
```
Nice job on sourcing environment variables from the relevant config.

bootupAbdullah committed:
Merge branch 'main' into test/twilio-sms

bootupAbdullah commented:
✅ All conflicts resolved in server.js and messaging.routes.js
✅ Kept proper route prefixes and corrected variable names
✅ Final push complete — ready for merge 🚀

bootupAbdullah commented:
✅ Merge conflicts resolved in server.js and messaging.routes.js
✅ Twilio SMS tested and confirmed working
✅ Updated config usage per review
🔁 Requesting re-review to unblock merge

echelonnought reviewed (changes requested):
Hey man @bootupAbdullah, I see this here 

``` 
"packageManager": "pnpm@10.11.1+sha512.e519b9f7639869dc8d5c3c5dfef73b3f091094b0a006d7317353c72b124e80e1afd429732e28705ad6bfa1ee879c1fce46c128ccebd3192101f43dd67c667912"
```
from the package.json file.
I think this need not to be, it probably got added when you ran `pnpm install`.

antaerus commented on send-test-sms.js:
```diff
@@ -0,0 +1,17 @@
+  })
+  .catch(error => {
+    console.error('Error sending SMS:', error.message);
+  });
```
With this PR, it's clear you've successfully verified that SMS sending works, which was the core goal of this issue.

I have some concerns about including this executable script in the repo that I'd like you to consider:

The main issue: This script sends real SMS messages every time it runs, which means:
- Future developers might run it accidentally while exploring the codebase
- It could be executed multiple times during testing/CI, leading to unnecessary SMS charges
- Someone could potentially use it to send spam messages

A better approach for this PR:
As @SAUMILDHANKAR mentioned before, documentation that shows other developers how to test Twilio integration locally could be better. Something like:
markdown

## Testing Twilio SMS Integration

### Setup
1. Add your test credentials to `.env`:
TWILIO_ACCOUNT_SID=your_test_sid
TWILIO_AUTH_TOKEN=your_test_token
TWILIO_PHONE_NUMBER=your_test_number

### Manual Testing (Local Development Only)

```javascript
// Example code developers can copy/paste into a temporary file
const twilio = require('twilio');
const { twilioAccountSID, twilioAuthToken, twilioPhoneNumber } = require('./src/config/config');

const client = twilio(twilioAccountSID, twilioAuthToken);

let response;

try {
   response = await client.messages.create({
       body: 'Test message',
       from: twilioPhoneNumber,
       to: 'your_phone_number'
   });
   logger.log("Message sent! SID:', message.sid);
} catch (error) {
   logger.error("Error sending "SMS:", error.message);
}

```

antaerus replied:
This way, developers can still test the integration when needed, but it requires intentional action rather than accidental execution.

**Why this matters for future development:**
- Prevents accidental costs from SMS sends
- Keeps the repo focused on production code
- Makes it clear that SMS sending should be intentional, not accidental

Such documentation changes could be done in a separate PR. Curious as to @ramakanth98's thoughts on this

antaerus replied:
TL;DR - Great that you've verified that your Twilio credentials are working, though probably better to remove this file.

antaerus replied:
@bootupAbdullah - were you able to read these comments?

bootupAbdullah replied:
@peanutenthusiast 

Great call on this! You're absolutely right about the risks of having executable SMS scripts in the repo. I've made the requested changes:

- Removed the executable script
- Added comprehensive documentation (TWILIO_TESTING.md) showing how to test SMS integration locally
- Included safety guidelines to prevent accidental execution and unnecessary charges

Thanks for the thoughtful feedback - this approach is much safer for the team.

antaerus commented on .docker/Dockerfile:
```diff
@@ -6,6 +6,7 @@ RUN corepack use pnpm@*
 
 ARG WORKDIR=/chatbot_server
 WORKDIR ${WORKDIR}
+ADD scripts/ ${WORKDIR}/scripts/
```
I'm not sure this line is needed? I can run `./scripts/rebuild.sh` without it.

bootupAbdullah replied:
Removed the ADD scripts/ line from the Dockerfile as initially requested.

After testing, it was confirmed it's required — without it, Docker can't run chmod during postinstall due to the missing scripts/ directory.

Re-added the line to restore expected behavior and allow containers to build and start properly.

antaerus replied:
My other PR removes [the postinstall chmod line](https://github.com/OpenSourceFellows/chatbot/pull/45/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L27), since it's already referenced [in the Dockerfile](https://github.com/OpenSourceFellows/chatbot/blob/main/.docker/Dockerfile#L20)

antaerus replied:
@bootupAbdullah - without this line, are you able to run the dev container? and/or execute `./scripts/rebuild.sh`?

antaerus reviewed (changes requested):
Requested changes:
1. Check whether you can execute `./scripts/rebuild.sh` successfully without this below line:
```ADD scripts/ ${WORKDIR}/scripts/```
2. Concurrent with @SAUMILDHANKAR's thoughts, documentation on how to verify successful SMS send might be more helpful than adding a js file at the project root.

bootupAbdullah committed:
Remove ADD scripts line as requested

bootupAbdullah committed:
Restore ADD scripts line to fix missing folder in container

bootupAbdullah commented:
@echelonnought Hey — I wasn’t able to tell which line or file you were referencing in your last comment. Could you point it out so I can take another look? Appreciate it.
![Screenshot 2025-07-08 at 8 10 58 PM](https://github.com/user-attachments/assets/f85c3ccd-ddb4-4fea-9bf3-aca87c9b2ec8)

echelonnought commented:
Yeah apologies about that, I edited that comment. That line comes from the package.json file.

bootupAbdullah committed:
Remove packageManager line from package.json

bootupAbdullah commented:
@echelonnought hey, the packageManager line was removed. Please let me know if anything else is needed. Thanks!

bootupAbdullah committed:
Replace executable SMS script with testing documentation

- Remove test script to prevent accidental SMS sends and charges
- Add comprehensive guide for local Twilio SMS testing
- Document safe practices for SMS integration verification

bootupAbdullah commented:
Changes Made:

✅ Removed the executable test-twilio.js script from the repository

✅ Created comprehensive documentation (TWILIO_TESTING.md) with safe testing guidelines

✅ Fixed the package.json line as requested in your follow-up comment

The documentation now guides developers to create temporary test scripts outside the repo directory, ensuring SMS functionality can be verified without the security and cost risks.

echelonnought reviewed (approved):
LGTM

bootupAbdullah committed:
Merge branch 'main' into test/twilio-sms

bootupAbdullah committed:
Remove ISSUE_TEMPLATE folder from test/twilio-sms

bootupAbdullah committed:
resolved merge conflicts in pnpm-lock.yaml

bootupAbdullah commented:
@SAUMILDHANKAR 

Deleted the unintended file and resolved the merge conflict in pnpm-lock.yaml. Everything should be clean now.

SAUMILDHANKAR commented on .github/PULL_REQUEST_TEMPLATE.md:
@bootupAbdullah My understanding is that if we merge this PR this file is going to get deleted which is not what we want. Adding a commit which reverts your earlier commit might be what you need. So, for anyone reviewing this PR, this file should not be in the files changed tab.

SAUMILDHANKAR commented:
> @SAUMILDHANKAR
> 
> Deleted the unintended file and resolved the merge conflict in pnpm-lock.yaml. Everything should be clean now.

Congratulations on fixing the merge conflict. Great awesome work. Very well done.

antaerus reviewed (commented):
Appreciate your updated changes and the documentation for testing Twilio functionality. I asked one question about Dockerfile setup, though this is mostly non-blocking.
